processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -758,10 +758,11 @@ public String getNull() {
             return "0";
         }
         if ( "char".equals( getName() ) ) {
-            return "'\\u0000'";
+            //"'\u0000'" would have been better, but depends on platform encoding
+                return "0";
         }
         if ( "double".equals( getName() ) ) {
-            return "0.0";
+            return "0.0d";
```

Comment:
Well nothing failed. The other scenario where this was used is the `NestedPropertyMappingMethod.ftl`.